Rebase branch on top of recent upstream master

Vtec234 opened the pull request:
I'm sorry

TheJJ commented:
Hm, it looks like you rebased way more than just the renderer branch. Did you do `git checkout TheJJ-renderer` and then `git rebase upstream/master`?

Vtec234 commented:
Okay, I fixed it, not sure what went wrong. This time it does _not_ include patches to make it build/work, but I don't want to mess with the mercy of git, so if you merge this I can submit a PR with the fix.

TheJJ commented:
It is still way too many commits in that pullreq, you didn't rebase correcly i fear.
I just made sure `TheJJ:master` is even with `SFTtech:master`, maybe that was part of the problem?

Vtec234 commented:
No no, now there are as many commits as there should be. If you look at the renderer branch, it's "29 commits ahead, 184 commits behind SFTtech:master". Since a rebase puts the renderer commits on top of the new SFTtech:master commits, it includes all 184 upstream commits and the 29 ones from this branch which look the same but are actually new since they now patch different upstream ones. 184+29=213, so the amount is correct. Also, while previously diffing this (my renderer branch = this PR) with the upstream master resulted in >400 commits, now there are only the 29 ones from this branch, so the rebase is correct.

TheJJ commented:
Makes sense, i never took a rebase from somebody else so far. What commits did you change for the rebase?
Also, there are still conflicts with the branch to current master. Didn't you encounter them when rebasing?
edit: i failed, the rebase requires a forcepush of course.

Vtec234 commented:
I fixed the build, and the renderer definitely works. Not sure about the main engine, but anyhow the renderer should be rebased before working on merging it with the new GUI.
EDIT: I also modified my commits so that they use my GitHub email. It's still mergeable fine into upstream master and still has to be force merged into this fork.
